src/Checkout/ExpressCheckout/ExpressCheckoutSubscriber.php

```diff
@@ -101,8 +102,17 @@ public function addExpressCheckoutDataToPage(PageLoadedEvent $event): void
             || $event instanceof NavigationPageLoadedEvent
             || $event instanceof SearchPageLoadedEvent;
 
+        if ($event instanceof ProductPageLoadedEvent) {
+            $event->getSalesChannelContext()->addExtension(
+                self::PAYPAL_PAYLATER_PRODUCT,
+                $event->getPage()->getProduct(),
+            );
+        }
+
         $expressCheckoutButtonData = $this->getExpressCheckoutButtonData($event->getSalesChannelContext(), $event::class, $addProductToCart);
```

Comment:
trunk is v10 for 6.7, how about we do the break there to pass the product instead of the bool `$addProductToCart` and just do this ugly Extension for 6.6.x and 6.5.x?